src/nci_util.c

```diff
@@ -849,6 +849,105 @@ nci_parse_intf_activated_ntf(
     return FALSE;
 }
 
+static
+gsize nci_mode_param_inplace_copy(const NciModeParam* param, NciModeParam* copy,
+    NCI_MODE mode)
```

Comment:
Let's use memcpy-style parameter order (destination first, source second):
```c
static
gsize
nci_mode_param_copy_impl(
    NciModeParam* dest,
    const NciModeParam* src,
    NCI_MODE mode)
```
(I also didn't quite understand what `inplace` meant in this context)